packages/typescript-estree/src/withoutProjectParserOptions.ts

```diff
@@ -8,12 +8,15 @@ import type { TSESTreeOptions } from './parser-options';
  *
  * @see https://github.com/typescript-eslint/typescript-eslint/issues/8428
  */
-export function withoutProjectParserOptions(
-  opts: TSESTreeOptions,
-): TSESTreeOptions {
+export function withoutProjectParserOptions<Options extends TSESTreeOptions>(
```

Comment:
You could make this even more generic
The user needn't have some subtype of our options and should be able to pass any object in really. This function just culls the properties. Whether or not they exist is irrelevant!


```suggestion
export function withoutProjectParserOptions<Options extends object>(
```